perf/reader.go

```diff
@@ -423,6 +429,5 @@ func (uev *unknownEventError) Error() string {
 // IsUnknownEvent returns true if the error occured
 // because an unknown event was submitted to the perf event ring.
 func IsUnknownEvent(err error) bool {
-	_, ok := errors.Cause(err).(*unknownEventError)
-	return ok
+	return xerrors.Is(err, (err).(*unknownEventError))
```

Comment:
I think I mentioned this before, this will cause a panic if err is not an unknownEventError. Have a look at https://golang.org/ref/spec#Type_assertions

```suggestion
	var uee *unknownEventError
	return xerrors.As(err, &uee)
```